app/spectrum.c

```diff
@@ -870,13 +870,16 @@ uint8_t Rssi2Y(uint16_t rssi)
 #ifdef ENABLE_FEAT_F4HWN_SPECTRUM
     static void DrawSpectrum()
     {
+        uint16_t steps = GetStepsCount();
+        uint16_t bars = (steps > 128) ? 128 : steps;
+        uint16_t shift_graph = 64 / steps + 1; // to center bar on freq marker
```

Comment:
i don't know which is more valuable: calculation time or storage. maybe these could be inlined to save space, but I think the render may be faster avoiding the recalculation.